src/iam.ts

```diff
@@ -272,6 +277,27 @@ export class IAM extends IAMBase {
     throw new Error(ERROR_MESSAGES.DID_DOCUMENT_NOT_INITIALIZED);
   }
 
+  async updateAssetDidDocument(options: {
```

Comment:
Method signature is too vague. I suggest something more descriptive like 
`addAssetToDelegates(assetDid, key?) { ... } `

 `IAM._publicKey` is sensible default value for key

Reply:
May be I am wrong, but my thinking behind this is that purpose of this PR is to approve asset to act on behalf of its owner. Propsed signature more precisely reflect this and abstract from details of the delegation mechanism (client dosn't need to know about document)

Reply:
@JGiter Thanks for elaborating 🙇  That's not quite right though. The purpose is add a publicKey to asset DID document so that the keypair can act **as the asset**.